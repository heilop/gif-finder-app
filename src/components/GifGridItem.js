import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({id, title, url}) => {

  return (
    <div className = 'card animate_animated animate_fadeIn'>
      <img src={ url } alt={ title }></img>
      <p>{ title }</p>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
