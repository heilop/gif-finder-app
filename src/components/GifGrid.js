import React from 'react';
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs'
// import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ( { category }) => {

  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{ category }</h3>
      { loading && <p className="card-grid animate__animated animate__flash">Loading</p> }

      <div className="card-grid animate__animated animate__fadeIn">

        {
          images.map(img => (
            <GifGridItem
              key = { img.id }
              img = {img}
              { ...img }
            />
          ))
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
