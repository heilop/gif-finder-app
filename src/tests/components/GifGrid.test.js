import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Test in <GifGrid /> component', () => {

  const category = 'Rick and Morty';

  test('should show the GifGrid component', () => {

    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    })

    const wrapper = shallow(<GifGrid category = { category } />);
    expect(wrapper).toMatchSnapshot();
  })

  test('should show items when images are loaded useFetchGifs', () => {

    const gifs = [{
      id: 'ABC',
      url: 'http://localhost/demo/image.jpg',
      title: 'Image Title',
    }]
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    })

    const wrapper = shallow(<GifGrid category = { category } />);

  })

});
