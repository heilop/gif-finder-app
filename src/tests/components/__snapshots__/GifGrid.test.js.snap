// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGrid /> component should show the GifGrid component 1`] = `
<Fragment>
  <h3>
    Rick and Morty
  </h3>
  <p
    className="card-grid animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid animate__animated animate__fadeIn"
  />
</Fragment>
`;
